File: oop/abstractClass/after/character.py
from abc import ABC, abstractmethod

class Character(ABC):

  @abstractmethod
  def __init__(self, name, hp, strength, defense):
    self.__name = name
    self.__hp = hp
    self.__strength = strength
    self.__defense = defense

  def getName(self):
    return self.__name

  def getHp(self):
    return self.__hp

  def getStrength(self):
    return self.__strength

  def getDefense(self):
    return self.__defense

  def setName(self, name):
    self.__name = name

  def setHp(self, hp):
    self.__hp = hp

  def setStrength(self, strength):
    self.__strength = strength

  def setDefense(self, defense):
    self.__defense = defense

  @abstractmethod
  def calculateDmg(self):
    pass

  def __str__(self):
    return f'{self.__name}: hp={self.__hp}, strength={self.__strength}, defense={self.__defense}'

File: oop/abstractClass/java/Character.java
package com.cgtfarmer.example;

abstract class Character {

  protected String name;
  protected int strength;

  private int hp;
  private int defense;

  public Character(String name, int hp, int strength, int defense) {
    this.name = name;
    this.hp = hp;
    this.strength = strength;
    this.defense = defense;
  }

  public String getName() {
    return this.name;
  }

  public int getHp() {
    return this.hp;
  }

  public int getStrength() {
    return this.strength;
  }

  public int getDefense() {
    return this.defense;
  }

  public void setHp(int hp) {
    this.hp = hp;
  }

  public void setStrength(int strength) {
    this.strength = strength;
  }

  public void setDefense(int defense) {
    this.defense = defense;
  }

  public float calculateDmg() throws Exception {
    throw new Exception("Must override this method");
  }

  // public void thingy() {
  //   int x = 5;
  //   System.out.println(x);

  //   String name = "John";
  //   System.out.println(name);

  //   String fullName = name + " Doe";
  //   System.out.println(fullName);
  // }

  public String toString() {
    return "name=" + this.name + " hp=" + this.hp + " strength=" + this.strength + " defense=" + this.defense;
  }
}
